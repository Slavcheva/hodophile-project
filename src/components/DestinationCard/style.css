.destinationCard {
    margin: 2em 1em;
    font-size: 0.5em;
    height: auto;
    width: 100%;
}

.destinationCard img {
    position: relative;
    overflow: hidden;
    width: 100%;
    border-radius: 0.5em;
    background-color: transparent;
}

.destinationCard img:before {
    display: block;
    content: '';
    padding-top: 56.25%;
}

.destinationCard img img {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 100%;
    height: auto;
    display: block;

}

.destinationCard .card-header {
    position: absolute;
    padding: 10px;
    width: 100%;
    margin: 0;
    color: black;
    z-index: 1;
}

.destinationCard .card-author {
    color: rgba(0, 0, 0, 0.44);
}

