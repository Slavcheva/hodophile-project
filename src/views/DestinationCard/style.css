@import url('https://fonts.googleapis.com/css2?family=Cormorant:ital,wght@1,600&display=swap');

.card-wrapper{
    width: 33%;

    height: auto;
    margin: 0 auto;
    padding: 3em ;
    text-align: center;
    border-radius: 1em;
    position: relative;
    display: inline-block;
}
.card-wrapper a{
    background-color: transparent;

}
.card-container{
    width: 100%;
    position: relative;
    margin: 5px;
    height: auto;
    overflow: hidden;
    /*background-color: black;*/
}
.card-container img{
    display: block;
    width: 100%;
    height: auto;
    opacity: 1;
    /*max-width: 430px;*/
}
.card-header{
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    margin: 0;
    font-weight: 100;
    margin-block-end: 1em;
    margin-block-start: 1em;
    margin-inline-end: 0;
    margin-inline-start: 0;
    color: white;
    font-size: 1em;
}

/*div .destination-wrapper p {*/
/*    display: flex;*/
/*    flex-wrap: wrap;*/
/*    justify-content: center;*/
/*}*/

/*div .welcome {*/
/*    padding: 10px;*/
/*    box-shadow: 0 5px 10px 5px #cecece;*/
/*}*/

/*.destination {*/
/*    padding: 10px;*/
/*    box-shadow: 0 5px 10px 5px #cecece;*/
/*    font: 30px/1.5 'Cormorant', cursive;*/
/*}*/

/*img {*/
/*    max-width: 200px;*/
/*    height: auto;*/
/*    margin: 0 auto;*/
/*}*/

