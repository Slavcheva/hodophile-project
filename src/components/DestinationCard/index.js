import React from 'react';

import './style.css'
import {Link} from "react-router-dom";

function DestinationCard({...trip}) {

    return <div className="card-wrapper">
        <div className='card-container'>
            <Link to={`/trip/${trip._id}`}>
                <h5 className='card-header'>{trip.destination}</h5>
                <div>
                <img src={trip.imageUrl} alt={trip.imageAlt}/>
                </div>
                <span className='card-author'><small>Author: </small>{trip.author}</span>
            </Link>
        </div>
    </div>;
}


export default DestinationCard;