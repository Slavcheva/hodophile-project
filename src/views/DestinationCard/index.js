import React from 'react';
import './style.css'
import {Link} from "react-router-dom";

function DestinationCard({...trip}) {

    return <div className="card-wrapper">
        <Link to={`/trip/${trip._id}`}>
            <div className='card-container'>
                <h5 className='card-header'>{trip.destination}</h5>
                <img src={trip.imageUrl} alt={trip.imageAlt}/>
                <span><small>Author: </small>{trip.author}</span>
            </div>
        </Link>
    </div>;
}


export default DestinationCard;