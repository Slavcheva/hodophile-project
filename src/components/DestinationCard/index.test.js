import React from "react";
import renderer from 'react-test-renderer'
import DestinationCard from "./index";

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('DestinationCard component', () => {
    test('should render correctly', () => {
        const component = renderer.create(
            <DestinationCard/>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})