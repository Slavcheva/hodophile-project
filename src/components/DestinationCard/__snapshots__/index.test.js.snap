// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationCard component should render correctly 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card-container"
  >
    <Link
      to="/trip/undefined"
    >
      <h5
        className="card-header"
      />
      <div>
        <img />
      </div>
      <span
        className="card-author"
      >
        <small>
          Author: 
        </small>
      </span>
    </Link>
  </div>
</div>
`;
